packages/material-ui/src/internal/SwitchBase.test.js

```diff
@@ -207,9 +207,10 @@ describe('<SwitchBase />', () => {
       );
 
       getByRole('checkbox').click();
+      setProps({ checked: false });
```

Comment:
We're relaying on a reference to the target now. If the `checked` prop is not updated, `event.target.checked` will be true again in next tick.

Reply:
The `click()` was missing an `act()` i.e. flushing of effects. The `setProps` was implicitly doing that.

The other problem is that the test was no longer testing the `click()` but rather `setProps({ checked: false })`. I changed the test so that we actually test `onChange={(event) => setChecked(event.target.checked)}`